src/charm.py

```diff
@@ -439,7 +439,7 @@ def _common_exit_hook(self, update_ca_certs: bool = False) -> None:
             self.unit.status = BlockedStatus(str(e))
             return
 
-        if update_ca_certs:
+        if update_ca_certs and self.server_cert.enabled:
```

Comment:
This hook can also be triggered to unlink the certificates after the relation to CA is removed. With this change we wouldn't be unlinking the CA cert which we likely want to do.